### ISSUE TITLE
한줄씩 파이토치 Part.2

### ISSUE THREAD

**Issue by Chromatic-Hwi** [Chromatic-Hwi/PyTorch-New#4]
![image](https://github.com/Chromatic-Hwi/PyTorch-New/assets/89522246/4a1395f2-cdf4-4a16-93d2-2fde4e152a7b)

- 입력층은 변수의 개수만큼, 출력층은 예측 변수의 개수만큼 Node를 갖기 때문에 데이터에 의해 정해지지만, 은닉층의 Layer와 Node 개수는 네트워크를 설계하면서 직접 설정해야 함.

- 다층 퍼셉트론은 순전파(Feed Forward)와 역전파(Back Propagation)을 반복하면서 가중치를 추정.

- 순전파 : 가중치 값을 랜덤으로 주고 입력층에서 출력층의 방향으로 단순 연산.

- 역전파 : 출력층에서 입력층의 방향으로 오차를 줄일 수 있게 가중치를 조정.

- Epoch : 갖고 있는 모든 데이터에 대해 Feed Forward와 Back Propagation을 한 번식 진행한 것.

- 퍼셉트론은 특정 임계치 이상이 되면 신호가 전달되는 Step Function을 사용했지만 인공신경망에서는 가중치를 찾는 과정에서 미분이 필요하므로 임계치 부분에서는 미분이 불가능한 Step Function을 사용할 수 없음. Sigmoid Function으로 해결하려 했지만 **Gradient Vanishing" 문제가 발생했고, 이를 극복하고자 미분 가능한 여러 활성화 함수를 사용.

- Softmax Function : 입력값을 0에서 1사이로 정규화하여 출력값의 총합을 항상 1로 만듬. 

# 

- CNN : 이미지를 1차원 벡터로 변환해 입력 받았던 다층 퍼셉트론의 문제를 이미지 그대로 입력을 받아 해결한 모델. 회귀와 분류에 모두 적용할 수 있음.

- CNN 구조
  - Convolutional Layer : 이미지의 공간적 정보를 학습.
  - Pooling Layer : Convolution Layer의 차원 형태 크기를 줄여 학습 Parameter 개수 감소.
  - FC Layer : 최종 출력을 위한 다층 퍼셉트론 구조

- 출력함수는 마지막 출력층의 결과를 목적에 맞는 적절한 형태로 변형해주는 역할을 함. 회귀 문제일때는 입력값을 그대로 출력하는 Identify Function, 분류 문제면 Softmax Function을 사용.

- Convolutional Layer : 이미지의 중요한 지역 정보를 뽑는 역할을 함. 이미지 필터(Kernel)를 적용하여 합성곱 연산 수행.

- Pooling Layer : Convolutional Layer로 계산된 Feature Map의 크기를 줄여 연산량을 줄이는 역할. Convolutional Layer와 달리 단순 계산만 진행하기 때문에 학습할 가중치는 따로 없음. 

- FC Layer : 최종 출력값은 분류 문제이면 각 클래스에 대한 예측 확률 벡터, 회귀 문제면 예측값 벡터를 출력. FC Layer는 Flatten Layer를 통과한 벡터를 출력값의 길이로 변환시키는 역할.
